components/resource_metering/src/reporter/single_target.rs

```diff
@@ -94,24 +116,93 @@ impl DataSink for SingleTargetDataSink {
                 warn!("failed to receive from a grpc call"; "error" => ?err);
             }
         });
+    }
+
+    fn update_data_sink_and_client(&mut self, new_address: String) {
+        if new_address.is_empty() {
+            self.reset();
+            return;
+        }
+
+        if self.address != new_address {
+            self.address = new_address;
+            let channel = {
+                let cb = ChannelBuilder::new(self.env.clone())
+                    .keepalive_time(Duration::from_secs(10))
+                    .keepalive_timeout(Duration::from_secs(3));
+                cb.connect(&self.address)
```

Comment:
Previously this is called at TrySend, but now it is called when address is changed. Will it cause problems when connect takes long time?

Reply:
I will move it to `try_send`.